js/modules/k6/data/share_test.go

```diff
@@ -43,22 +45,33 @@ var array = new data.SharedArray("shared",function() {
 });
 `
 
-func newConfiguredRuntime(moduleInstance interface{}) (*goja.Runtime, error) {
+func newConfiguredRuntime() (*goja.Runtime, error) {
 	rt := goja.New()
 	rt.SetFieldNameMapper(common.FieldNameMapper{})
-	ctx := common.WithRuntime(context.Background(), rt)
-	err := rt.Set("data", common.Bind(rt, moduleInstance, &ctx))
+
+	m, ok := New().NewModuleInstance(
+		&modulestest.VU{
+			RuntimeField: rt,
+			InitEnvField: &common.InitEnvironment{},
+			CtxField:     common.WithRuntime(context.Background(), rt),
+			StateField:   nil,
+		},
+	).(*Data)
+	if !ok {
+		return rt, fmt.Errorf("not a Data module instance")
+	}
```

Comment:
Perhaps, not even a request change, but a curiosity question: why didn't you use `require.True(t, ok)` here?

Reply:
assert|require with concurrency could generate a race so I tend to avoid it for helper functions if it's possible (if the func return error). Otherwise, I would prefer a classic `require.True` as you suggested.

Reply:
I think what you are talking about @codebien is that `t.FailNow` and co. are [supposed](https://pkg.go.dev/testing#B.FailNow) to be called *only* from the goroutine that is running the test, not by anyone that is spawned by it. 

We do ... not do that quite a lot it just so happens that in most cases the failure never happens :smile:  
I think it's perfectly fine to leave the `error` instead of using `require.True` and co., but also in this case I would expect it's perfectly fine to use it as well

Reply:
>that in most cases the failure never happens
Sure, but in case of changes for refactoring, you could get the race.

>but also in this case I would expect it's perfectly fine to use it as well

Yep, but I don't know how the helper could be used in the next test we will write. For this reason, I think it makes sense just for helpers shared across multiple tests. Ofc, in a single test I would prefer to use directly the assert function.